src/main/java/com/jherrick/Agent.java

```diff
@@ -9,5 +9,5 @@ public interface Agent {
      * @param board - the game board
      * @return the move that the Agent will make
      */
-    Move pickMove(Board board);
+    Move pickMove(BigBoard board);
```

Comment:
As much as possible, all dependencies should be on the Board interface, not on the BigBoard implementation. The value added from interfaces is breaking the dependency on specific implementations. If there's a gap in the Board interface that makes evaluating them impossible, we should add new methods to the Board interface to change that.

Reply:
Got it that makes totally sense. I'm guessing by having "BigBoard" as the parameter it defeats the purpose of having a Board interface, and doesn't allow us to hot swap other Boards into our code (in case we wanted a BigBoard with 2D Array. 

I think I got confused on whether the Board interface referred to BigBoards, SubBoards, or Both. I'm going to assume BigBoard for now, I don't really think that it referring to both would be that useful. There are a few similarities, but a lot more larger differences that they tend to act differently.